internal/cluster/cluster_test.go

```diff
@@ -578,6 +578,162 @@ var _ = Describe("lease timeout event", func() {
 	})
 })
 
+var _ = Describe("Auto assign machine CIDR", func() {
+	var (
+		db          *gorm.DB
+		c           common.Cluster
+		id          strfmt.UUID
+		clusterApi  *Manager
+		ctrl        *gomock.Controller
+		mockHostAPI *host.MockAPI
+		mockMetric  *metrics.MockAPI
+		dbName      = "cluster_monitor"
+		mockEvents  *events.MockHandler
+	)
+	BeforeEach(func() {
+		db = common.PrepareTestDB(dbName, &events.Event{})
+		id = strfmt.UUID(uuid.New().String())
+		ctrl = gomock.NewController(GinkgoT())
+		mockHostAPI = host.NewMockAPI(ctrl)
+		mockMetric = metrics.NewMockAPI(ctrl)
+		mockEvents = events.NewMockHandler(ctrl)
+		dummy := &leader.DummyElector{}
+		clusterApi = NewManager(defaultTestConfig, getTestLog().WithField("pkg", "cluster-monitor"), db,
+			mockEvents, mockHostAPI, mockMetric, dummy)
+	})
+	tests := []struct {
+		name                    string
+		srcState                string
+		machineNetworkCIDR      string
+		expectedMachineCIDR     string
+		hosts                   []*models.Host
+		eventCallExpected       bool
+		userActionResetExpected bool
+	}{
+		{
```

Comment:
Add a cluster with an already assigned machine cidr scenario

Reply:
+1

Reply:
First one was checked.
Will add test for second one.